packages/x-charts/src/ChartsText/ChartsText.tsx

```diff
@@ -24,14 +25,17 @@ function ChartsText(props: ChartsTextProps) {
 
   const { angle, textAnchor, dominantBaseline, ...style } = styleProps ?? {};
 
+  const isClient = useIsClient();
+
   const wordsByLines = React.useMemo(
-    () => getWordsByLines({ style, needsComputation: text.includes('\n'), text }),
-    [style, text],
+    () => getWordsByLines({ style, needsComputation: isClient && text.includes('\n'), text }),
+    [style, text, isClient],
   );
 
   let startDy: number;
   switch (dominantBaseline) {
     case 'hanging':
+    case 'text-before-edge':
```

Comment:
I adde dthis one and the  `text-after-edge` because it gives better result.

Makes me realise the `hanging` and `auto` are not 100% correctly computed but nobody complained about it